### PR TITLE
Add cctag 1.0.4

### PR DESCRIPTION
**Pull request by EstebanDugueperoux2**
### Summary
Changes to recipe:  **lib/[version]**

#### Details

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by valgur — APPROVED** [review]

**Comment by EstebanDugueperoux2** [comment]
Hi @jcar87 there seems to have a CI issue, how Can WE fix this missing prebuilt packagé issue?

Regards.

File: recipes/cctag/all/conanfile.py [hunk 1]
```diff
@@ -126,10 +126,11 @@ def _patch_sources(self):
         replace_in_file(self, os.path.join(self.source_folder, "src", "CMakeLists.txt"),
                               "${OpenCV_LIBS}",
                               "opencv_core opencv_videoio opencv_imgproc opencv_imgcodecs")
-        # Remove very old CUDA compute capabilities
-        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
-                              "set(CCTAG_CUDA_CC_LIST_INIT0 3.5 3.7 5.0 5.2)",
-                              "set(CCTAG_CUDA_CC_LIST_INIT0 5.0 5.2)")
+        if Version(self.version) < "1.0.4":
+            # Remove very old CUDA compute capabilities
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
```
Comment (perseoGI) [inline recipes/cctag/all/conanfile.py:131]: Hello @EstebanDugueperoux2, the PR title describes a new `cctag` version but here you are also altering the recipe.
Could you shed some light on why it is no longer necessary to remove these files?
It is very recommended in this cases to add a comment describing why this changes are not necessary anymore.

Comment (EstebanDugueperoux2) [reply recipes/cctag/all/conanfile.py:132]: Hello @perseoGI I have added a comment about https://github.com/alicevision/CCTag/pull/210/files
Tell me if sufficient.

**Review by perseoGI — DISMISSED** [review]
LGTM

File: recipes/cctag/all/conanfile.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
+        # boost/1.85.0 not compatible because of "error: 'numeric' is not a namespace-name" error
         self.requires("boost/1.84.0", transitive_headers=True, transitive_libs=True)
```
Comment (uilianries) [inline recipes/cctag/all/conanfile.py:63]:
```suggestion
        if Version(self.version) >= "1.0.4":
            self.requires("boost/1.84.0", transitive_headers=True, transitive_libs=True)
        else:
            # boost/1.85.0 not compatible because of "error: 'numeric' is not a namespace-name" error
            self.requires("boost/1.84.0", transitive_headers=True, transitive_libs=True)
```

It seems like the new patch version fixed that problem: https://github.com/alicevision/CCTag/compare/v1.0.3...v1.0.4#diff-d975bf659606195d2165918f93e1cf680ef68ea3c9cab994f033705fea8238b2R30

Comment (AbrilRBS) [reply recipes/cctag/all/conanfile.py:63]: Great catch!

**Review by uilianries — APPROVED** [review]
LGTM. It looks simple a patch version, but has many fixes that affected the build. 

https://github.com/alicevision/CCTag/compare/v1.0.3...v1.0.4#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

**Review by perseoGI — APPROVED** [review]
